### PR TITLE
Validate no white space in the password

### PR DESCRIPTION
**Pull request by vkghub**
[Pivotal-180388495](https://www.pivotaltracker.com/n/projects/2502930/stories/180388495)

This is a back end validation that will prevent users from saving passwords with blank spaces in it

We are also working on a front-end (JavaScript-based) validation for this so that users can get the immediate feedback on password validity

### CODE REVIEW

File: app/models/concerns/authorization_concern.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ module AuthorizationConcern
     # field :confirmation_sent_at, type: Time
     # field :unconfirmed_email,    type: String # Only if using reconfirmable
 
+    validates :password, format: { without: /\s/, message: "Password must not contain spaces" }
```
Comment (nisanthyaganti9) [inline app/models/concerns/authorization_concern.rb:43]: Unblocking comment -

How about including this in `password_complexity` def itself. And, this method needed a refactor, it's doing same checks multiple times.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -134,6 +134,14 @@ def is_shop_or_fehb_market_enabled?
       end
     end
 
+    context 'when password has white spaces' do
+      let(:params){valid_params.deep_merge!({password: "Test@1234 "})}
```
Comment (nisanthyaganti9) [inline spec/models/user_spec.rb:138]: I believe we have to allow "Test@1234 " by squishing input to remove whitespaces around it but, should be blocking "Test @1234". what do you think?
(I posted this on story to get answer from business)

File: app/models/concerns/authorization_concern.rb (hunk 1 above)
Comment (vkghub) [reply app/models/concerns/authorization_concern.rb:43]: We will be using a separate ticket for code refactor.

File: spec/models/user_spec.rb (hunk 2 above)
Comment (vkghub) [reply spec/models/user_spec.rb:138]: As per the discussion, we should not be allowing the user to enter white spaces. in password

**Review by nisanthyaganti9 — APPROVED** [review]

**Review by markgoho — CHANGES_REQUESTED** [review]
please hold off merging this

not allowing spaces in passwords is not a modern security practice

**Review by markgoho — APPROVED** [review]

**Pull request merged**